force_bdss/app/evaluate_operation.py

```diff
@@ -37,5 +41,8 @@ def run(self):
         mco_data_values = mco_communicator.receive_from_mco(mco_model)
 
         kpi_results = self.workflow.execute(mco_data_values)
-
         mco_communicator.send_to_mco(mco_model, kpi_results)
```

Comment:
As you suggested, @sparsonslab - I think it would be a good idea to place these lines in a try-except-finally statement, similar to the `OptimizeOperation`

Reply:
I have tried issue #317, but see my latest comments.

Whatever one does, it becomes a soup of unit test errors which aren't at all obvious to me.

Reply:
I saw in your comment, sounds nasty. 

If this is something you would like to experiment with / keep separate from the current PR then feel free to make a new branch off `base-operation-listeners` with your implementation and then create a PR with it that uses `base-operation-listeners` as the base branch.

That way you can share ideas / ask for review whilst keeping this PR clean.